server/server.js

```diff
@@ -342,6 +346,151 @@ app.post('/message', jsonBodyParser, async (req, res) => {
     }
 });
 
+app.post('/heartbeat', jsonBodyParser, async (req, res) => {
+    log(`got a heartbeat from ${req.body.system_id}, flic_last_seen_secs is ${req.body.flic_last_seen_secs}, flic_last_ping_secs is ${req.body.flic_last_ping_secs}`)
+    let hub = await db.getHubWithSystemId(req.body.system_id)
+    hub.flicLastSeenTime = moment().subtract(req.body.flic_last_seen_secs, 'seconds').toISOString()
+    hub.flicLastPingTime = moment().subtract(req.body.flic_last_ping_secs, 'seconds').toISOString()
+    hub.heartbeatLastSeenTime = moment().toISOString()
+    await db.saveHeartbeat(hub)
+    res.status(200).send()
+})
+
+
+app.post('/heartbeat/rename_system', jsonBodyParser, async (req, res) => {
+    log('got a request to rename system ' + req.body.system_id)
+    let hub = await db.getHubWithSystemId(req.body.system_id)
+    hub.systemName = req.body.system_name
+    await db.saveHubRename(hub)
+    res.status(200).send()
+})
+
+app.post('/heartbeat/hide_system', jsonBodyParser, async (req, res) => {
+    log('got a request to hide system ' + req.body.system_id) 
+    let hub = await db.getHubWithSystemId(req.body.system_id)
```

Comment:
I don't think you need to make this DB call. Instead of passing the whole Hub object into `db.saveHubHideStatus`, could you just send in the `true` or `false` instead?

Same for unhide, rename, mute, unmute, and heartbeat